backend/src/main/java/com/bakdata/conquery/models/worker/Namespaces.java

```diff
@@ -77,14 +83,16 @@ public CentralRegistry getMetaRegistry() {
 	}
 
 	public synchronized void register(SlaveInformation slave, WorkerInformation info) {
-		WorkerInformation old = workers.getOptional(info.getId()).orElse(null);
+
+		final WorkerInformation old = get(info.getDataset()).getWorkers().get(info.getId());
+
 		if (old != null) {
 			old.setIncludedBuckets(info.getIncludedBuckets());
 			old.setConnectedSlave(slave);
 		}
 		else {
 			info.setConnectedSlave(slave);
-			workers.add(info);
+			getWorkers().add(info);
```

Comment:
Öhm, `getWorkers` alloziiert eine map, gib die zurück, hier wird etwas geadded und dann geht sie in den GC?